controllers/events.ts

```diff
@@ -120,4 +125,40 @@ const getCalendarEvents = async (req: Request, res: Response): Promise<any> => {
   }
 };
 
-export { postEvent, getEvents, getCalendarEvents };
+type TAcknowledgementStatusReq = z.infer<typeof getAcknowledgementSchema>;
+
+const getAcknowledgementStatus = async (
+  req: Request,
+  res: Response
+): Promise<Response> => {
+  const { eventId, attendeeId } = req.params;
+  const { status } = req.body as TAcknowledgementStatusReq['body'];
+
+  const attendeeData = await prisma.attendees.findFirst({
+    where: {
+      eventId: Number(eventId),
+      attendeeId: Number(attendeeId),
+    },
+  });
+
+  if (attendeeData === null) {
+    return res.boom(Boom.badRequest('Attendee not found'));
+  }
+
+  const updatedData = await prisma.attendees.update({
+    where: {
+      id: attendeeData.id,
+    },
+    data: {
+      acknowledgement: status,
+    },
+  });
+
+  if (updatedData && updatedData.acknowledgement === status) {
```

Comment:
can we do `updatedData?.acknowledgment` instead of  `updatedData && updatedData.acknowledgement === status` ?

Reply:
This would be true if we are checking that acknowledgement exists in `updatedData` as optional chaining returns undefiened for the item we are trying to access incase it's not present. 

Looking at this now I feel doing the `updatedData && updated...` check is not really necessary as the `update` function from prisma will always return us an attendee and just checking `updatedData.acknowledgement` should be fine.